pkg/webhook/external/generic_list_options.go

```diff
@@ -0,0 +1,186 @@
+package external
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/pkg/expressions"
+)
+
+func listTargetObjects(
```

Comment:
This is another spot where it might be cleaner to make this function a method of the `genericWebhookReceiver` type instead of passing the client around.